hip-clang.docker

```diff
@@ -28,6 +28,7 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-
     libpthread-stubs0-dev \
     llvm-amdgpu \
     miopengemm \
+    ninja-build \
```

Comment:
Why is this necessary?

Reply:
`ninja-build` is the default (and faster) way to build LLVM project. If you are against pulling in this item, I can explore to try using cmake only.

Reply:
But I believe rocm just uses make to build the llvm package.

Reply:
I didn't know about rocm way to build llvm package. If I understand it right, there is an option to use cmake -G with different generators. ninja are the one we typically use because it is faster. I can explore the -G Unix Makefiles option.